### PR TITLE
Enable and fix a number of additional lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -9,33 +9,54 @@ analyzer:
...
     - directives_ordering
     - empty_statements
+    - file_names
     - implementation_imports
     - iterable_contains_unrelated_type
     - list_remove_unrelated_type
     - literal_only_boolean_expressions
+    - missing_whitespace_between_adjacent_strings
     - no_adjacent_strings_in_list
+    - no_runtimeType_toString
```
Comment (natebosch) [inline analysis_options.yaml]: Why do we care about this one?

File: lib/src/solver/version_solver.dart [hunk 2]
```diff
@@ -185,6 +185,7 @@ class VersionSolver {
     // [incompatibility] is satisfied and we have a conflict.
     if (unsatisfied == null) return #conflict;
 
+    // ignore: missing_whitespace_between_adjacent_strings
```
Comment (natebosch) [inline lib/src/solver/version_solver.dart]: Is there a bug in the lint? Have you filed an issue?

File: lib/src/utils.dart [hunk 3]
```diff
@@ -241,7 +241,8 @@ bool isLoopback(String host) {
 
   try {
     return InternetAddress(host).isLoopback;
-  } on ArgumentError catch (_) {
+  } on ArgumentError catch (_) // ignore: avoid_catching_errors
```
Comment (natebosch) [inline lib/src/utils.dart]: 😞 

https://github.com/dart-lang/sdk/issues/40692

File: lib/src/solver/version_solver.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/solver/version_solver.dart]: mostly for JS compile. 🤷‍♂

File: analysis_options.yaml (hunk 1 above)
Comment (kevmoo) [reply analysis_options.yaml]: mostly for JS compile. 🤷‍♂

File: lib/src/solver/version_solver.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/solver/version_solver.dart]: No bug. In this case, the lint is not valid – because we don't need whitespace here

Comment (natebosch) [reply lib/src/solver/version_solver.dart]: Right, so we should file a bug on the lint.

Comment (kevmoo) [reply lib/src/solver/version_solver.dart]: No. The lint is doing the right thing. It makes sure that

```dart
"This is my long sentence error message"
"that's missing a space"
```

is correct. But you get false positives.

Comment (kevmoo) [reply lib/src/solver/version_solver.dart]: fixed

**Comment by kevmoo** [comment]
@natebosch – LGTY?

**Pull request merged**